user/tests/wiring/serial_loopback2/loopback.cpp

```diff
@@ -19,36 +19,59 @@
 #include "unit-test/unit-test.h"
 #include "random.h"
 
-test(SERIAL_00_LoopbackNoDataLossAndAvailableIsCorrect) {
-    const size_t TEST_BUFFER_SIZE_MIN = 8;
-    const size_t TEST_BUFFER_SIZE_MAX = SERIAL_BUFFER_SIZE / 2;
-    const unsigned ITERATIONS = 10000;
-    const unsigned BAUD_RATE = 115200;
-
+void runLoopback(size_t buffer_size_min, size_t buffer_size_max) {
     particle::Random rand;
 
+    size_t bufferSize = random(buffer_size_min, buffer_size_max);
+    // Generate random data
+    char txBuf[bufferSize + 1] = {};
+    rand.genBase32(txBuf, bufferSize);
+
+    Serial1.write(txBuf);
+    Serial1.flush();
+
+    size_t pos = 0;
+    char rxBuf[bufferSize] = {};
+    do {
+        size_t available = bufferSize - pos;
+        assertEqual(available, Serial1.available());
+        if (available) {
+            rxBuf[pos] = (char)Serial1.read();
+        }
+    } while (++pos <= bufferSize);
+
+    assertTrue(!strncmp(txBuf, rxBuf, bufferSize));
+}
+
+test(SERIAL_00_LoopbackNoDataLossAndAvailableIsCorrect) {
```

Comment:
Another test case should be added that any leftover data in the receive buffer is left untouched between the sleep/wake.